### ISSUE TITLE
Unseen categorical values in test

### ISSUE THREAD

**Issue by WalidMustapha** [catboost/catboost#507]
How does Catboost transform unseen categorical values? I see why the value will default to `prior`, but what are the implications of this for prediction?

This [issue](https://github.com/catboost/catboost/issues/18) touches on the topic but was closed before a clear answer was provided. 

I have conducted some prediction experiments using seen and unseen categorical values, but I'm unable to infer what Catboost actually does just by looking at the associated predictions. Often, the prediction using the unseen corresponds to the prediction associated with the most frequently seen value, but this not always the case.

The [documentation](https://tech.yandex.com/catboost/doc/dg/concepts/algorithm-main-stages_cat-to-numberic-docpage/) is unfortunately not clear on this particular topic.

Are you able to elaborate?

**Comment by annaveronika** [comment]
You are correct, prior is used if categorical feature value has not been seen in train. Here is the paper that contains full description of how categorical features are handled: http://learningsys.org/nips17/assets/papers/paper_11.pdf

Documentation also contains the needed information. It's pretty long, if it's not clear enough, I'm not sure I can write it more clear one more time in the issue. If you read the paper and the docs one more time, and get a good idea how to write it more clear in documentation, please write to us, we'll update.

**Comment by annaveronika** [comment]
there are also several videos, that could be helpful for understanding categorical features handling https://tech.yandex.com/catboost/doc/dg/concepts/educational-materials-videos-docpage/

I would recommend to watch High level overview of the library and Ordered boosting and categorical features processing in CatBoost

**Comment by annaveronika** [comment]
I hope this helps. We do as much as we can in explaining the algorithm, but still get many questions. If anything is not clear enough, please try to understand it - we have it all in docs, papers and videos. And then write to us, how to improve our docs. This will help a lot to all users of the library.

**Comment by WalidMustapha** [comment]
Thanks for your input @annaveronika 

A little time spent reading the paper clarified the issue for me. Especially the following was illuminating:

> "For regression tasks standard technique for calculating prior is to take the average label value in
> the dataset. For binary classification task a prior is usually an a priori probability of encountering
> a positive class"

I'm not certain whether this belongs in the documentation or if having it in the paper is sufficient. Perhaps most users are comfortable knowing unseen values are handled correctly rather than knowing the implementation details, so I guess this is more a matter of preference. 

Thank you again :-)

**Comment by HarjyotKaur** [comment]
Hey
Thanks for sharing the videos. I understand prior is used in case of target encoding, but what happens in the case of one-hot encoding, do all the one-hot encoding columns becomes zero for an unseen categorical value?
